recovery/multiimagewritethread.cpp

```diff
@@ -719,26 +720,29 @@ bool MultiImageWriteThread::untar(const QString &tarball)
 {
     QString cmd = "sh -o pipefail -c \"";
 
-    if (isURL(tarball))
+    QString tarballPath = QString(tarball);
+    if (isURL(tarball)){
+        tarballPath = QUrl::fromUserInput(tarball).toString(QUrl::RemoveQuery);
         cmd += "wget --no-verbose --tries=inf -O- "+tarball+" | ";
```

Comment:
Might need to escape shell arguments if you are going to allow query parameters.
While ? does not have special meaning to shell, other characters that may appear such as & do.